### PR TITLE
Elastic unbounded range query

### LINKED ISSUE #498: Support for elastic search syntax for unbounded RangeQuery
**Is your feature request related to a problem? Please describe.**
See #328 
QUoting 

> I like this option from [ElasticSearch syntax]
> (https://www.elastic.co/guide/en/elasticsearch/reference/6.3/query-dsl-query-string-query.html#_ranges):
> 
> `field:>=5`
> `field:>2`
> `field:<3`
> `field:<=12`

Currently we  support `foo:[* TO whatever]` or `foo:[whatever TO *]`

### PR DESCRIPTION
**Pull request by petr-tik**
Been working on this for a couple of days. Find parser generators tricky, so still walking through it in the debugger. 

Took time to tidy-up and add comments, where I thought it would be useful.

Had a separate elastic_range parser! function first, thought it would be better for unit testing. Ended up merging its functionality into parser! range function, but finding it hard to debug. 

Closes #498

### CODE REVIEW

**Comment by fulmicoton** [comment]
@petr-tik can you fix the unit test?

File: src/query/query_parser/query_grammar.rs
Comment (fulmicoton) [inline src/query/query_parser/query_grammar.rs]: The absence of fields is handled below.

If your change works, let's remove the now obsolete `term_default_field` below.
If it does not, let's remove your change.

**Comment by petr-tik** [comment]
Weirdly enough, travis reported the build as green, despite the failure in one of the tests

https://travis-ci.org/tantivy-search/tantivy/jobs/570252947#L2601

I think this error proves that our previous parser was inconsistent with assigning bounds and delimiters. 

I propose choosing the "{" as a delimiter for the unbounded range like "weight: >50" = "weight:{50 TO *}". Mathematically speaking an unbounded range doesn't have a limit value, therefore has to be exclusive, so we should use the exclusive delimiter for it.

File: src/query/query_parser/query_grammar.rs
Comment (petr-tik) [reply src/query/query_parser/query_grammar.rs]: reverted my change

**Comment by petr-tik** [comment]
GitHub doesn't give me a button to squash commits. Tried making it locally. 

Please review, but don't merge, I will work on squashing this into 1 commit tomorrow (it's past 1am in London)

**Comment by fulmicoton** [comment]
@petr-tik the default button for merging is "squash and merge"

**Comment by fulmicoton** [comment]
Can you run `cargo fmt`? I suspect you will have a few diffs here.

**Comment by petr-tik** [comment]
I run cargo-fmt locally (version below) and I didn't get any diffs, when I ran it before the patch. 

If you are referring to the indentation in the parser functions - I did this on purpose, because I think it helps legibility. 

```bash
$$$ cargo-fmt --version
rustfmt 1.2.2-stable (5274b49 2019-04-24)
```

Please review and approve (re-requested), so I can squash and merge (the button is currently greyed out)

File: src/query/query_parser/query_grammar.rs
Comment (fulmicoton) [inline src/query/query_parser/query_grammar.rs]: i am not fond of the position of the second `}`. 

I'd put that  on a new line.

**Comment by fulmicoton** [comment]
@petr-tik  i'm not fond of the second `}` on l.121.
It is very hard to see what it's `{` counterpart is.

File: src/query/query_parser/query_grammar.rs
Comment (petr-tik) [reply src/query/query_parser/query_grammar.rs]: I see your point, moved the last "}" of both such if-elses to respective new lines

**Comment by fulmicoton** [comment]
@petr-tik thanks

**Comment by petr-tik** [comment]
Are you happy with the rest of the PR to approve it, so I can squash-merge?

**Comment by fulmicoton** [comment]
@petr-tik I don't think you have permission to merge. I will squash-merge as soon as travis is done.

**Comment by petr-tik** [comment]
I think I might have such permissions (the merge request buttons were green in other PRs), but I need an approving review. 

Let me know how you prefer to do this

**Comment by fulmicoton** [comment]
I'm ok if you want to merge it.

**Review by fulmicoton — APPROVED** [review]

**Comment by fulmicoton** [comment]
I approved the changes.

**Comment by petr-tik** [comment]
You are right, I cannot merge this PR. 

Accessing this PR from my phone (logged in) suggests that I can, but I guess I don't have write access to the repo, which keeps the button greyed out from the web UI

**Pull request merged**
